accounts/kv_sql_migration_test.go

```diff
@@ -0,0 +1,339 @@
+package accounts
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/lightninglabs/lightning-terminal/db"
+	"github.com/lightningnetwork/lnd/clock"
+	"github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/sqldb"
+	"github.com/stretchr/testify/require"
+)
+
+// TestAccountStoreMigration tests the migration of account store from a bolt
+// backed to a SQL database. Note that this test does not attempt to be a
+// complete migration test.
+func TestAccountStoreMigration(t *testing.T) {
+	t.Parallel()
+
+	ctxc, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	// First create a shared Postgres instance so we don't spawn a new
+	// docker container for each test.
+	pgFixture := db.NewTestPgFixture(
+		t, db.DefaultPostgresFixtureLifetime, true,
+	)
+	t.Cleanup(func() {
+		pgFixture.TearDown(t)
+	})
+
+	makeSQLDB := func(t *testing.T, clock *clock.TestClock,
+		sqlite bool) (*SQLStore, *db.TransactionExecutor[SQLQueries]) {
+
+		var store *SQLStore
+
+		if sqlite {
+			store = NewSQLStore(db.NewTestSqliteDB(t).BaseDB, clock)
+		} else {
+			store = NewSQLStore(
+				db.NewTestPostgresDB(t).BaseDB, clock,
+			)
+		}
+
+		baseDB := store.BaseDB
+
+		genericExecutor := db.NewTransactionExecutor(
+			baseDB, func(tx *sql.Tx) SQLQueries {
+				return baseDB.WithTx(tx)
+			},
+		)
+
+		return store, genericExecutor
+	}
+
+	migrationTest := func(t *testing.T, kvStore *BoltStore,
+		clock *clock.TestClock, sqlite bool) {
+
+		sqlStore, txEx := makeSQLDB(t, clock, sqlite)
+
+		var opts sqldb.MigrationTxOptions
+		err := txEx.ExecTx(
+			ctxc, &opts, func(tx SQLQueries) error {
+				return MigrateAccountStoreToSQL(
+					ctxc, kvStore, tx,
+				)
+			},
+		)
+		require.NoError(t, err)
+
+		// MigrateAccountStoreToSQL will check if the inserted accounts
+		// and indices equals the migrated ones, but as a sanity check
+		// we'll also fetch the accounts and indices from the store and
+		// compare them to the original.
+		// First we compare the migrated accounts to the original ones.
+		kvAccounts, err := kvStore.Accounts(ctxc)
+		require.NoError(t, err)
+		numAccounts := len(kvAccounts)
+
+		sqlAccounts, err := sqlStore.Accounts(ctxc)
+		require.NoError(t, err)
+		require.Equal(t, numAccounts, len(sqlAccounts))
+
+		for i := 0; i < numAccounts; i++ {
+			assertEqualAccounts(t, kvAccounts[i], sqlAccounts[i])
+		}
+
+		// After that we compare the migrated indices.
+		kvAddIndex, kvSettleIndex, err := kvStore.LastIndexes(ctxc)
+		if errors.Is(err, ErrNoInvoiceIndexKnown) {
+			// If the db doesn't have any indices, we can't compare
+			// them.
```

Comment:
if the kvstore didnt populate during test set up though, then we will silently exit here. Rather have a test param like "expectLastIndex" so that we know for sure if we are getting the right result here

Reply:
Ah good catch, I actually forgot to store the indexes when i added invoices, so that caught that :rocket:!